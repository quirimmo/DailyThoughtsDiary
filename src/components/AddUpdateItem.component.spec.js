import React from 'react';
import NavigationTestUtils from 'react-navigation/NavigationTestUtils';
import renderer from 'react-test-renderer';
import moment from 'moment';

jest.mock('NativeAnimatedHelper');

import AddUpdateItem from './AddUpdateItem.component';

const mockOnSubmitItem = jest.fn();
const item = {
	symptoms: 'symptoms',
	location: 'location',
	thoughts: 'thoughts',
	date: '2018-10-28T22:00:46.723Z'
};
let component;
let instance;

describe('AddUpdateItem', () => {
	beforeEach(() => {
		jest.useFakeTimers();
		NavigationTestUtils.resetInternalState();
		component = renderer.create(
			<AddUpdateItem
				currentItem={item}
				onSubmitItem={mockOnSubmitItem}
				buttonTitle="button-title"
			/>
		);
		instance = component.getInstance();
	});
	afterEach(() => {
		instance.setState({ currentItem: instance.state.currentItem });
		jest.clearAllMocks();
	});

	it('should be defined', () => {
		expect(AddUpdateItem).toBeDefined();
	});

	it('should render the component', async () => {
		expect(component.toJSON()).toMatchSnapshot();
	});

	describe('state', () => {
		it('should init the currentItem property', () => {
			expect(instance.state.currentItem).toEqual(item);
		});
	});

	describe('isSubmitDisabled', () => {
		it('should return false', () => {
			expect(instance.isSubmitDisabled()).toBeFalsy();
		});

		it('should return true', () => {
			instance.setState({
				currentItem: { ...instance.state.currentItem, symptoms: '' }
			});
			expect(instance.isSubmitDisabled()).toBeTruthy();
		});

		it('should return true', () => {
			instance.setState({
				currentItem: { ...instance.state.currentItem, location: '' }
			});
			expect(instance.isSubmitDisabled()).toBeTruthy();
		});

		it('should return true', () => {
			instance.setState({
				currentItem: { ...instance.state.currentItem, thoughts: '' }
			});
			expect(instance.isSubmitDisabled()).toBeTruthy();
		});

		it('should return true', () => {
			instance.setState({
				currentItem: { location: '', thoughts: '', symptoms: '' }
			});
			expect(instance.isSubmitDisabled()).toBeTruthy();
		});
	});

	describe('changeItemText', () => {
		it('should change the property of the currentItem in the state with the given value', () => {
			instance.changeItemText('testProperty', 'testValue');
			expect(instance.state.currentItem.testProperty).toEqual('testValue');
		});
	});

	describe('onSetTime', () => {
		it('should set the date property of the currentItem in the state', () => {
			instance.onSetTime(10, 20);
			expect(moment('2018-10-28T10:20:46.723').isSame(instance.state.currentItem.date)).toBeTruthy();
		});
	});

	describe('onSubmitItem', () => {
		it('should call the onSubmitItem prop', () => {
			instance.onSubmitItem();
			expect(mockOnSubmitItem).toHaveBeenCalledWith(instance.state.currentItem);
		});
	});
});
