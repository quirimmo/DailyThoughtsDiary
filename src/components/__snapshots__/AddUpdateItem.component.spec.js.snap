// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddUpdateItem should render the component 1`] = `
<View>
  <View
    style={
      Object {
        "marginBottom": 10,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={undefined}
    >
      Symptoms:
    </Text>
    <TextInput
      allowFontScaling={true}
      onChangeText={[Function]}
      placeholder="Type symptoms..."
      style={
        Object {
          "fontSize": 16,
          "height": 50,
        }
      }
      value="symptoms"
    />
  </View>
  <View
    style={
      Object {
        "marginBottom": 10,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={undefined}
    >
      Location:
    </Text>
    <TextInput
      allowFontScaling={true}
      onChangeText={[Function]}
      placeholder="Type where you are..."
      style={
        Object {
          "fontSize": 16,
          "height": 50,
        }
      }
      value="location"
    />
  </View>
  <View
    style={
      Object {
        "marginBottom": 10,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={undefined}
    >
      Thoughts:
    </Text>
    <TextInput
      allowFontScaling={true}
      onChangeText={[Function]}
      placeholder="Type thoughts..."
      style={
        Object {
          "fontSize": 16,
          "height": 50,
        }
      }
      value="thoughts"
    />
  </View>
  <View
    style={
      Object {
        "marginBottom": 10,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "fontSize": 18,
        }
      }
    >
      Time:
    </Text>
    <View
      style={
        Object {
          "flexDirection": "row",
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "flex": 0.5,
            "flexDirection": "row",
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "fontSize": 16,
            }
          }
        >
          23:00
        </Text>
      </View>
      <View
        style={
          Object {
            "flex": 0.5,
          }
        }
      >
        <View
          accessibilityComponentType="button"
          accessibilityLabel={undefined}
          accessibilityTraits={
            Array [
              "button",
            ]
          }
          accessible={true}
          collapsable={undefined}
          hasTVPreferredFocus={undefined}
          hitSlop={undefined}
          isTVSelectable={true}
          nativeID={undefined}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
          testID={undefined}
          tvParallaxProperties={undefined}
        >
          <View
            style={
              Array [
                Object {},
              ]
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              disabled={undefined}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {
                    "color": "#007AFF",
                    "fontSize": 18,
                    "padding": 8,
                    "textAlign": "center",
                  },
                ]
              }
            >
              Change Time
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "marginTop": 30,
      }
    }
  >
    <View
      accessibilityComponentType="button"
      accessibilityLabel={undefined}
      accessibilityTraits={
        Array [
          "button",
        ]
      }
      accessible={true}
      collapsable={undefined}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <View
        style={
          Array [
            Object {},
          ]
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          disabled={false}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "#007AFF",
                "fontSize": 18,
                "padding": 8,
                "textAlign": "center",
              },
            ]
          }
        >
          button-title
        </Text>
      </View>
    </View>
  </View>
</View>
`;
